Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/// <reference path="../../definitions/node.d.ts"/>
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/shelljs.d.ts"/>
+
+import tl = require('vsts-task-lib/task');
+import fs = require('fs');
+import path = require('path');
+import shell = require('shelljs');
+import Q = require('q');
+
+var request = require('request');
+
+class Credential {
+    mUsername: string;
+    mPassword: string;
+
+    constructor(username: string, password: string) {
+        this.mUsername = username;
+        this.mPassword = password;
+    }
+}
+
+function getRequest(url: string, cred: Credential, strictSSL: boolean, callback: (res) => void): void {
+    request
+        .get({url: url, strictSSL: strictSSL}, (err, res, body) => {
+            if (res && body && res.statusCode === 200)  {
+                callback.call(this, JSON.parse(body));
+            } else {
+                if (res && res.statusCode) {
+                   tl.debug(tl.loc('ServerCallErrorCode', res.statusCode)); 
+                }
+                if (body) {
+                    tl.debug(body);
+                }
+                tl.setResult(tl.TaskResult.Failed, tl.loc('ServerCallFailed'));
+            }
+        })
+        .auth(cred.mUsername, cred.mPassword, true)
+        .on('error', err => {
+            tl.setResult(tl.TaskResult.Failed, err);
+        });
+}
+
+function getLastSuccessful(serverEndpointUrl: string, jobName: string, cred: Credential, strictSSL: boolean): Q.Promise<number> {
+    let defer = Q.defer<number>();
+    let lastSuccessfulUrl = `${serverEndpointUrl}/job/${jobName}/api/json?tree=lastSuccessfulBuild[id,displayname]`;
+
+    getRequest(lastSuccessfulUrl, cred, strictSSL, result => {
```

Comment:
The promise handling isn't quite right; if you hit an error in getRequest(), the callback will never be called, and the promise will never be fulfilled, and then you will be stuck